docs/en/react/react-lynx-testing-library/index.mdx

```diff
@@ -0,0 +1,404 @@
+    // we are using random timeout here to simulate a real-time example
+    // of an async operation calling a callback at a non-deterministic time
+    const randomTimeout = Math.floor(Math.random() * 100);
+
+    setTimeout(() => {
+      resolve({ returnedMessage: 'Hello World' });
+    }, randomTimeout);
+  });
+
+class ComponentWithLoader extends Component {
+  state = { loading: true };
+
+  componentDidMount() {
+    fetchAMessage().then((data) => {
+      this.setState({ data, loading: false });
+    });
+  }
+
+  render() {
+    if (this.state.loading) {
+      return <text>Loading...</text>;
+    }
+
+    return (
+      <text data-testid="message">
+        Loaded this message: {this.state.data.returnedMessage}!
+      </text>
+    );
+  }
+}
+
+test('it waits for the data to be loaded', async () => {
+  render(<ComponentWithLoader />);
+  // elementTree.root in Lynx Environment is used to maintain the page element tree
+  expect(elementTree.root).toMatchInlineSnapshot(`
+    <page>
+      <text>
+        Loading...
+      </text>
+    </page>
+  `);
+  const loading = () => {
+    return screen.getByText('Loading...');
+  };
+  await waitForElementToBeRemoved(loading);
+  // Since Lynx Environment uses jsdom to implement Element PAPI at the bottom layer
+  // you can directly access document.body to get page elements
+  expect(document.body).toMatchInlineSnapshot(`
+    <body>
+      <page>
+        <text
+          data-testid="message"
+        >
+          Loaded this message:
+          <wrapper>
+            Hello World
+          </wrapper>
+          !
+        </text>
+      </page>
+    </body>
+  `);
+  expect(screen.getByTestId('message')).toHaveTextContent(/Hello World/);
+  expect(elementTree.root).toMatchInlineSnapshot(`
+    <page>
+      <text
+        data-testid="message"
+      >
+        Loaded this message:
+        <wrapper>
+          Hello World
+        </wrapper>
+        !
+      </text>
+    </page>
+  `);
+});
+```
+
+### Rerender
+
+```jsx
+import '@testing-library/jest-dom';
+import { render } from '@lynx-js/react-lynx-testing-library';
+import { expect } from 'vitest';
+
+it('rerender will re-render the element', async () => {
+  const Greeting = (props) => <text>{props.message}</text>;
+  const { container, rerender } = render(<Greeting message="hi" />);
+  expect(container).toMatchInlineSnapshot(`
+    <page>
+      <text>
+        hi
+      </text>
+    </page>
+  `);
+  expect(container.firstChild).toHaveTextContent('hi');
+
+  {
+    // Unlike React Testing Library, container needs to be retrieved after rerender
+    // Because ReactLynx creates a new page element each time it loads
+    const { container } = rerender(<Greeting message="hey" />);
+    expect(container.firstChild).toHaveTextContent('hey');
+
+    expect(container).toMatchInlineSnapshot(`
+      <page>
+        <text>
+          hey
+        </text>
+      </page>
+    `);
+  }
+});
+```
+
+### More usage
```

Comment:
`<list/>` will be in "More usage", or should we guide here?

Reply:
I have added the list and mts example, PTAL again~